ci: update github runners to oci gh arc runners

tiagolobocastro opened the pull request:
CNCF has hosted ephemeral GitHub runners on Oracle that we want projects to use instead of the GitHub-hosted ones, which now incur a cost to use.

Please direct any questions to @jeefy, @krook, and @RobertKielty. (or #cncf-ci-infra channel on the CNCF Slack)

(Similar to https://github.com/openebs/openebs/pull/3914)

koksay committed:
ci: update github runners to oci gh arc runners

Signed-off-by: Koray Oksay <koray.oksay@gmail.com>

tiagolobocastro commented:
bors try

tiagolobocastro commented:
bors try

tiagolobocastro commented:
@koksay looks like we're running out of space during build.

oracle-vm-16cpu-64gb-x86-64:
```
 Filesystem      Size  Used Avail Use% Mounted on
tmpfs           6.3G  1.3M  6.3G   1% /run
/dev/sda1        77G   58G   20G  75% /
tmpfs            32G   84K   32G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sda16      881M   78M  742M  10% /boot
/dev/sda15      105M  6.2M   99M   6% /boot/efi
tmpfs           6.3G   12K  6.3G   1% /run/user/1000
```

ubuntu-latest-16-cores:
```
 Filesystem      Size  Used Avail Use% Mounted on
/dev/root       581G   61G  521G  11% /
tmpfs            32G   84K   32G   1% /dev/shm
tmpfs            13G  1.3M   13G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sda16      881M   69M  750M   9% /boot
/dev/sda15      105M  6.2M   99M   6% /boot/efi
tmpfs           6.3G   12K  6.3G   1% /run/user/1001
tmpfs           6.3G   12K  6.3G   1% /run/user/0
```

koksay commented:
@tiagolobocastro ok, I will update the vm shape later today and ping you

koksay commented:
@tiagolobocastro I've updated the VM configuration, can you retry?

```bash
df -h /
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda1       581G   47G  535G   8% /
```

tiagolobocastro commented:
bors try

Abhinandan-Purkait commented on .github/workflows/unit-int.yml:
```diff
@@ -9,7 +12,9 @@ env:
 jobs:
   int-tests:
-    runs-on: ubuntu-latest-16-cores
+    #runs-on: oracle-vm-16cpu-64gb-x86-64
+    #runs-on: ubuntu-latest-16-cores
```
Remove this?

Abhinandan-Purkait commented on .github/workflows/unit-int.yml:
```diff
@@ -111,7 +117,7 @@ jobs:
-      # - name: Setup tmate session
-      #   if: ${{ failure() }}
-      #   timeout-minutes: 240
-      #   uses: mxschmitt/action-tmate@v3
+      - name: Setup tmate session
+        if: ${{ failure() }}
+        timeout-minutes: 240
+        uses: mxschmitt/action-tmate@v3
```
Comment these?

tiagolobocastro replied:
No, I'm trying to debug the failures, it's not passing

Abhinandan-Purkait commented on test/python/common/nvme.py:
```diff
@@ -121,7 +121,7 @@ def nvme_find_ctrl(uri):
     # Finds correct Device
     devs = list(filter(lambda d: nqn in d.get("SubsystemNQN"), discover.get("Devices")))
-    assert len(devs) is 1, "Multiple devices with the same subnqn"
+    assert len(devs) == 1, "Multiple devices with the same subnqn"
```
What's the difference? `is` should also work right?

tiagolobocastro replied:
it's a warning with pytest

tiagolobocastro commented:
ok so **I think** it's failing because the /etc/nvme/{hostid,hostnqn} are not installed in this system; for a while now linux doesn't allow the same hostid to show up on a different hostnqn
I'll get this fixed

tiagolobocastro commented:
There's another failure, which fails on the new runner, and also btw on my system:
```
$> cargo test nexus_child_retire_persist_unresponsive_with_bdev_io --features io-engine-testing
```

tiagolobocastro commented:
> There's another failure, which fails on the new runner, and also btw on my system:
> 
> ```
> $> cargo test nexus_child_retire_persist_unresponsive_with_bdev_io --features io-engine-testing
> ```

Alright, IMHO this test cannot be written as is. It's writing directly to the bdev but this doesn't work correctly with retires because the pause/drain is only implemented for the nvmf subsystem, which means IO's aren't drained whilst we drop the handles.
So for now I'm inclined to allow this one to fail until we address this
CC @dsharma-dc

tiagolobocastro committed:
test: ignore test which is relying on absent behaviour

The test writes to the nexus bdev directly in a retire scenario where
frozen IOs are resubmitted.
The problem happens when we pause the nexus and drop the io channels
because pause only pauses the nvmf subsystem!!
So, if the IO which is resubmitted hasn't completed, then spdk will
assert because it expects IOs to be drained.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
test: extend fio completion timeout

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
test: use nvme path directly

Otherwise secure path may not allow us to call our nvme binary.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
test: ensure /etc/nvme is valid before running tests

Adds a dummy work around which checks if etc is configured, and otherwise
auto-generates or uses env variable.
Adds a script which can check and also write the contents. This second fix
is a more robust one, ensuring tests run in a good config.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
@koksay I think I've got this mostly ready now, one question though these runners with 16cpu actually seem to be 32 cpus?
bors try

tiagolobocastro commented:
bors try

jeefy commented:
It's a quirk of OCI VMs, it grants double the CPUs requested. :)

koksay commented:
Yes, as Jeefy said, Oracle Cloud has OCPU and VCPU. Each OCPU has 2 cores by default, so VCPU is double of the OCPU number. 

We preferred to use 16 OCPUs, so you see it as 32 on linux.

tiagolobocastro commented:
Alright, thank you.
Btw are there any VM ARM runners we can use? I had raised a ticket a while back and was given some ARM runners to test but hit the same issues we had hit with the non-vm runners previously.

After a bunch of fixes, looks like the tests pass now (and hopefully now more runner agnostic)
bors merge

koksay commented:
Arm runners are the next item on our to-do list. I will ping you when they are ready (hopefully in the next 2 weeks)

tiagolobocastro commented:
> Arm runners are the next item on our to-do list. I will ping you when they are ready (hopefully in the next 2 weeks)

Awesome, thank you!

tiagolobocastro commented:
@koksay looks like 1 workflow is stuck for 40 minutes waiting to start: https://github.com/openebs/mayastor/actions/runs/15605343469/job/43953425056
And the bdd worflow failed with:
```
The self-hosted runner lost communication with the server
```

koksay commented:
@tiagolobocastro can you retry the job? I was debugging what is happening (the VM actually starts and the parameters are sent, but somehow nothing happens)

tiagolobocastro commented:
bors try

tiagolobocastro commented:
Once more for luck :)
bors try

tiagolobocastro commented:
alright
bors merge
